drone_sar/ipsar_dataset.py

```diff
@@ -0,0 +1,88 @@
+import os
+import torch
+import logging
+import numpy as np
+from PIL import Image
+from torch.utils.data import Dataset
+from xml.etree import ElementTree
+from object_detector import ObjectDetector
+from utils import setup_logging
+
+
+class IPSARDataset(Dataset):
+    """
+    A PyTorch dataset for the IPSAR dataset.
+
+    Args:
+        root_dir (str): The path to the root directory of the IPSAR dataset.
+        set_name (str): The name of the dataset split ("train" or "test").
+    """
+
+    def __init__(self, root_dir: str, set_name: str):
+        assert set_name in ["train", "test"], f"Invalid set name: {set_name}"
+
+        self.images_path = os.path.join(root_dir, "heridal", set_name + "Images")
+        self.labels_path = os.path.join(self.images_path, "labels")
+        self.dataset = self._create_dataset()
+
+    def _create_dataset(self) -> Dataset:
+        dataset = []
+        for filename in os.listdir(os.path.join(self.images_path)):
+            if filename.endswith(".JPG"):
+                img_path = os.path.join(self.images_path, filename)
+                annotation_path = os.path.join(
+                    self.labels_path, filename.split(".")[0] + ".xml"
+                )
+
+                if os.path.exists(annotation_path):
+                    annotation = self._parse_annotation(annotation_path)
+                    if len(annotation["boxes"]) > 0:
+                        dataset.append(
+                            (
+                                np.asarray(Image.open(img_path)),
+                                {
+                                    "boxes": annotation["boxes"][0],
+                                    "class_labels": annotation["class_labels"][0],
+                                },
+                            )
+                        )
+                    else:
+                        # TODO: Add empty dataset entry
```

Comment:
We might not need explicit negative examples as stated here: https://ai.stackexchange.com/questions/28371/how-to-add-negative-samples-for-object-detection

> The paper basically proposes to think that each pixel of a dataset image is a training signal. Then for each image there are lots of pixels with negative signal (nothing in it: sky, ground, trees...) and only a few with positive signal (the actual car).

We can later verify if this is the case by predicting on the empty images and asserting there are not confidently wrong predictions.